core/src/commonMain/kotlin/behavior/EntitlementBehavior.kt

```diff
@@ -0,0 +1,60 @@
+package dev.kord.core.behavior
+
+import dev.kord.common.entity.Snowflake
+import dev.kord.common.exception.RequestException
+import dev.kord.core.entity.Entitlement
+import dev.kord.core.entity.KordEntity
+import dev.kord.core.entity.Strategizable
+import dev.kord.core.exception.EntityNotFoundException
+import dev.kord.core.supplier.EntitySupplier
+import dev.kord.rest.request.RestRequestException
+
+/**
+ * The behavior of a [Discord Test Entitlement](https://discord.com/developers/docs/monetization/entitlements)
+ */
+public interface EntitlementBehavior : KordEntity, Strategizable {
+    public val applicationId: Snowflake
+
+    /**
+     * Requests to get this value as an [Entitlement].
+     *
+     * @throws [RequestException] if anything went wrong during the request.
+     * @throws [EntityNotFoundException] if the entitlement wasn't present.
+     */
+    public suspend fun asEntitlement(): Entitlement = supplier.getEntitlement(applicationId, id)
+
+    /**
+     * Requests to get this value as an [Entitlement].
+     * returns null if this entitlement isn't present.
+     *
+     * @throws [RequestException] if anything went wrong during the request.
+     */
+    public suspend fun asEntitlementOrNull(): Entitlement? = supplier.getEntitlementOrNull(applicationId, id)
+
+    /**
+     * Retrieve the [Entitlement] associated with this behavior from the provided [EntitySupplier]
+     *
+     * @throws [RequestException] if anything went wrong during the request.
+     * @throws [EntityNotFoundException] if the entitlement wasn't present.
+     */
+    public suspend fun fetchEntitlement(): Entitlement = supplier.getEntitlement(applicationId, id)
+
+    /**
+     * Retrieve the [Entitlement] associated with this behavior from the provided [EntitySupplier]
+     * returns null if this entitlement isn't present.
+     *
+     * @throws [RequestException] if anything went wrong during the request.
+     */
+    public suspend fun fetchEntitlementOrNull(): Entitlement? = supplier.getEntitlementOrNull(applicationId, id)
```

Comment:
is the whole getting a single entitlement feature needed? do you have usecases in mind? it's not really documented after all, and you can always just [get multiple and filter](https://discord.com/developers/docs/monetization/entitlements#list-entitlements). would keep the api surface smaller too.

if we decide to drop it, we can probably drop `EntitlementBehavior` as a whole and just have `delete` as an (extension) function on `Entitlement`.

Reply:
I added them because they're present in other `Behavior` classes. I'm fine with dropping them 🤷🏼‍♂️

Reply:
i'd be in favor of dropping them then